### PR TITLE
Yml config replacers scope clash with config import

### PR DESCRIPTION
**Pull request by alex-dukhno**
Consequent PR for #15810 to fix scope clash between config replacers and imports in yaml files.
Should be rebased over master after #15810 will be merged

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/config/YamlConfigImportVariableReplacementTest.java [hunk 1]
```diff
@@ -587,8 +581,6 @@ public void testImportRedefinesSameConfigScalarThrows() throws Exception {
                 + "    - ${config.location}\n"
                 + "  cluster-name: name2";
 
-        rule.expect(new RootCauseMatcher(InvalidConfigurationException.class, "hazelcast/cluster-name"));
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/config/YamlConfigImportVariableReplacementTest.java]: I think we should keep this expectation here and below. Moving it to the `@Test` annotations removes the coverage for printing the YAML "path" led to the exception.

**Comment by blazember** [comment]
@alex-dukhno Feel free to @mention me if this PR is rebased on the XML counterpart and is ready to review

**Comment by alex-dukhno** [comment]
@blazember PR is ready for review. Thank you

**Comment by blazember** [comment]
@alex-dukhno thanks for updating the PR.
After checking it in local, I noticed some of the `YamlClientConfigImportVariableReplacementTest` cases fail. That's because of two things:
- The root node in the failing tests is `hazelcast:` instead of `hazelcast-client:`
- The test uses `HazelcastParallelClassRunner`, while the test cases write and read files. Some of them use files with the same name. When the tests executed in parallel, they may fail since parallel tests overwrite or delete the files the failing test tries to use. This was the case with the XML variant of the same test, see #16050 with some details and with the fix for that.

Another not obvious failure for the XML counterpart was detected by our Windows build job that was fixed in #16057. I can foresee the same failure in the YAML case.

Also, please make sure the test cases added by this PR are defined by the abstract parent class of the XML and YAML test classes and annotated with `@Override`. We do this to ensure we have the same coverage for the two types of the config.

Could you please take a look at these above?

**Comment by alex-dukhno** [comment]
@blazember could you please take a look one more time? Thank you.

**Comment by blazember** [comment]
run-lab-run

**Comment by blazember** [comment]
Thank you for your contribution @alex-dukhno :+1:  
Squashed and merged

**Pull request merged**